pkg/catalog/egress.go

```diff
@@ -136,8 +136,8 @@ func (mc *MeshCatalog) getUpstreamTrafficSettingForEgress(egressPolicy *policyv1
 				Namespace: egressPolicy.Namespace,
 				Name:      match.Name,
 			}
-			upstreamtrafficSetting := mc.policyController.GetUpstreamTrafficSetting(
-				policy.UpstreamTrafficSettingGetOpt{NamespacedName: &namespacedName})
+			upstreamtrafficSetting := mc.Interface.GetUpstreamTrafficSetting(
+				k8s.UpstreamTrafficSettingGetOpt{NamespacedName: &namespacedName})
```

Comment:
It's surprising to see the k8s pkg imported into this package.  My understanding was cataglog shouldn't know about k8s?  

Is this option parameter in the wrong place?

Reply:
Great point. I agree that `UpstreamTrafficSettingGetOpt` should remain in the policy package

Reply:
I might suggest the following:

1. place the entire `GetUpstreamTrafficSetting` method in compute/client.
2. Add 2 new methods to k8s.Client: `GetUpstreamTrafficSetting` (takes a name and namespace, no options), and `ListUpstreamTrafficSetting`
3. Break up the `GetUpstreamTrafficSetting` into 3 methods (all in compute.Client), one for each option. The options pattern isn't very well suited here, since combining/permuting options is meaningless.

The final result of all of these changes will be:

1. k8s/client.go: Is merged with the informer package, and contains simple CRUD operations for K8s objects
2. compute.Client, abstracts over the k8s.Client, and provides some logic for retrieving OSMs domain objects
3. MeshCatalog contains higher level methods for combining multiple objects to generate OSM domain objects (may eventually get merged with compute.Client

Reply:
Update:

1. `UpstreamTrafficSettingGetOpt` is removed.
2. `GetUpstreamTrafficSetting()` is split into 3 methods, one for each option. 

> compute.Client, abstracts over the k8s.Client, and provides some logic for retrieving OSMs domain objects

Do you mean compute.Client abstracts over kube.Client? Currently k8s.Client has no direct connection with compute.Client.

Reply:
the compute.Client has an embedded k8s.KubeController (which is a k8s.Client object)